app/helpers/repository_helper.rb

```diff
@@ -168,21 +168,17 @@ def remove_folders(folders, user, repo, path: '/', txn: nil)
         path = File.join obj.path, obj.name
         if obj.is_a? Repository::RevisionFile
           files << path
-        else
-          dirs[path] = current_revision
         end
       end
-      dirs[folder_path] = current_revision
+      dirs << folder_path
     end
-    success, file_messages = remove_files(files, user, repo, path: '', txn: txn)
+    success, file_messages = remove_files(files, user, repo, path: '', txn: txn, keep_folder: false)
 
     return [success, file_messages] unless success
 
     # folders are removed in git if their contents are removed
-    unless repo.is_a? GitRepository
-      dirs.each do |dir, revision|
-        txn.remove(dir, revision)
-      end
+    dirs.each do |dir|
```

Comment:
Please double check if the order that directories are deleted in matters or not for subversion and memory repositories.